docker.go

```diff
@@ -613,17 +616,39 @@ func (c *DockerContainer) CopyToContainer(ctx context.Context, fileContent []byt
 
 // StartLogProducer will start a concurrent process that will continuously read logs
 // from the container and will send them to each added LogConsumer
-func (c *DockerContainer) StartLogProducer(ctx context.Context) error {
-	if c.stopProducer != nil {
-		return errors.New("log producer already started")
+// timeout accepts values between 5 and 60s and will be used to set the context timeout
+// which means that each log-reading loop will last at least the specified timeout
+// and that it cannot be cancelled earlier
+func (c *DockerContainer) StartLogProducer(ctx context.Context, timeout time.Duration) error {
```

Comment:
Wdyt about this: we could probably create functional options for the StartLogProduce method:

```go
func (c *DockerContainer) StartLogProducer(ctx context.Context, opts ...LogProduceOptions) error
```

Being the timeout one of them. That would reduce the breaking change (BC) surface (only people assigning the function to a variable will get impacted). 

I'd try to avoid adding a BC here, but for this upcoming release we already have two (https://github.com/testcontainers/testcontainers-go/releases). So in the end I'm fine adding it if you consider it's fine too :)

Reply:
Yeah, I was already thinking about that. I think it's a good idea. Applied.